// Run
//
// python -m grpc_tools.protoc -I.  --python_out=. --pyi_out=. \
// --grpc_python_out=. ./bytecheckpoint/utilities/server/report_service.proto
//
// to generate new protos.

syntax = "proto3";

message ByteCheckpointGatherRequest {
  // Used to distinguish different tasks.
  string tag = 1;
  int32 rank = 2;
  bytes content = 3;
  bool with_result = 4;
}

message ByteCheckpointGatherResponse {
  repeated bytes contents = 1;
}

message ByteCheckpointBroadcastRequest {
  string tag = 1;
  int32 rank = 2;
  bytes content = 3;
  int32 src_rank = 4;
}

message ByteCheckpointBroadcastResponse {
  bytes content = 1;
}

message ByteCheckpointGetStatusRequest {
}

message ByteCheckpointGetStatusResponse {
  bytes status = 1;
}

service ByteCheckpointReportService {
  rpc Gather(ByteCheckpointGatherRequest) returns (ByteCheckpointGatherResponse) {
  }

  rpc Broadcast(ByteCheckpointBroadcastRequest) returns (ByteCheckpointBroadcastResponse) {
  }

  rpc GetStatus(ByteCheckpointGetStatusRequest) returns (ByteCheckpointGetStatusResponse) {
  }
}
